lib/matplotlib/backends/backend_agg.py

```diff
@@ -495,27 +494,11 @@ def print_png(self, filename_or_obj, *args,
             If the 'pnginfo' key is present, it completely overrides
             *metadata*, including the default 'Software' key.
         """
-
-        if metadata is None:
-            metadata = {}
-        if pil_kwargs is None:
-            pil_kwargs = {}
-        metadata = {
-            "Software":
-                f"matplotlib version{mpl.__version__}, http://matplotlib.org/",
-            **metadata,
-        }
         FigureCanvasAgg.draw(self)
-        # Only use the metadata kwarg if pnginfo is not set, because the
-        # semantics of duplicate keys in pnginfo is unclear.
-        if "pnginfo" not in pil_kwargs:
-            pnginfo = PngInfo()
-            for k, v in metadata.items():
-                pnginfo.add_text(k, v)
-            pil_kwargs["pnginfo"] = pnginfo
-        pil_kwargs.setdefault("dpi", (self.figure.dpi, self.figure.dpi))
-        (Image.fromarray(np.asarray(self.buffer_rgba()))
```

Comment:
We do `Image.fromarray` here, but `Image.frombuffer` below.  Can we skip a numpy conversion?

Reply:
Indeed, we can special-case this by just passing a memoryview around; done.